status-picker: Improve status picker styling.

shuklamaneesh23 opened the pull request:
Fixes: #35005.

<!-- Describe your pull request here.-->

Fixes: <!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

kuv2707 commented:
Looks fine to me at the code level. Can't comment on the appearance.

@zulipbot remove "buddy review"
@zulipbot add "mentor review"

shuklamaneesh23 commented:
Updates:

<img width="387" alt="Screenshot 2025-07-09 at 12 12 14 PM" src="https://github.com/user-attachments/assets/502d0853-8197-4002-94c2-bc4773debd1f" />
<img width="397" alt="Screenshot 2025-07-09 at 12 12 22 PM" src="https://github.com/user-attachments/assets/d8791122-01b0-4120-b104-10f4ce17a392" />

@alya please let me know if we need to update anything.
Thanks!

vrongmeal commented:
Looks good to me as well.

@zulipbot remove "mentor review"
@zulipbot add "maintainer review"

alya commented:
1. Do we have a delay on the emoji tooltip? `INTERACTIVE_HOVER_DELAY` would probably make sense here.
2. I think we should keep the purple highlighting around the emoji while the picker is open.

shuklamaneesh23 commented:
Hi @alya , updated the PR addressing your comment.
Please have another look.

<img width="407" height="196" alt="Screenshot 2025-07-21 at 11 44 26 PM" src="https://github.com/user-attachments/assets/0364e193-d034-4919-bdc9-79ecc146ad9f" />

- Added INTERACTIVE_HOVER_DELAY on the emoji tooltip.
- Kept the purple highlighting around the emoji while the picker is open.

timabbott commented on web/templates/set_status_overlay.hbs:
```diff
@@ -9,17 +9,21 @@
-            {{status_text}}
-        </button>
-    {{/each}}
+<div class="user-status-options popover-menu">
```
Abstractions wise, I would expect the `popover-menu` class to be on an outer div that is the entire popover, not have a `div` within the popover outside it. 

I bet this causes some sort of bug, but don't know precisely what.

@sayamsamal can you help get this cleaned up?

sayamsamal replied:
Sure thing, added to TODO list for today!

sayamsamal replied:
Yup, the CSS related to the `popover-menu` class is written in a way where it is expected that it would be the main parent div containing the `simplebar` divs inside it. So, things like the `border-radius` on it, and the `max-width` on the `simplebar-content` are some of the places which could negatively affect the modal.

sayamsamal commented:
I noticed that the "Select emoji" button is partially hidden, we should fix this.
<img width="405" height="128" alt="Screenshot 2025-07-29 at 1 15 16 AM" src="https://github.com/user-attachments/assets/12d2c635-42eb-4ef3-afeb-2c273955c1e7" />

sayamsamal commented:
Reading through the issue description and discussions at [#frontend > Clean up clickable lists in modals.](https://chat.zulip.org/#narrow/channel/6-frontend/topic/Clean.20up.20clickable.20lists.20in.20modals.2E), I feel it would be a better solution to just copy over the hover/focus styles from the popovers by either having them adjacent to one another or having them share common CSS variables. There is no need for us to force the use of `popover-menu` class which is supposed to be used in tandem with the `popover-menu` tippyJS theme anyways.
